rust/src/base.rs

```diff
@@ -101,43 +114,26 @@ impl Base {
         row: HashMap<String, String>,
         road_network: HaversineRoadNetwork,
     ) -> PyResult<Self> {
-        let base_id = match row.get(&"base_id".to_string()) {
-            Some(base_id) => base_id,
-            None => {
-                return Err(PyValueError::new_err(
-                    "cannot load base without a base_id value",
-                ))
-            }
-        }
-        .to_owned();
-        let lat_string = match row.get(&"lat".to_string()) {
-            Some(l) => l,
-            None => {
-                return Err(PyValueError::new_err(
-                    "cannot load base without a lat value",
-                ))
-            }
-        };
+        let base_id = row
```

Comment:
this is more concise. does this feel idiomatic? in general, do you feel we should be pushing toward the same kind of chaining that we get in Scala/Haskell with do notation / for comprehensions, where we aren't setting intermediate results to variables? or is that not really how folks write rust code?

like

```scala
for {
  base_id <- row.get("base_id").toRight(new Error("missing base id")
  lat_str <- row.get("lat").toRight(new Error("missing lat")
  ...
} yield Base(base_id, ...)
```
```

Reply:
tbh I'm still learning what is and isn't idiomatic but I think it might fall somewhere between what I have here and a pure functional notation (like Scala/Haskell)